lxd/networks.go

```diff
@@ -858,6 +863,77 @@ func networkLeasesGet(d *Daemon, r *http.Request) response.Response {
 	return response.SyncResponse(true, leases)
 }
 
+func networkGetLeaseAddresses(s *state.State, network string, hwaddr string) ([]api.InstanceStateNetworkAddress, error) {
+	addresses := []api.InstanceStateNetworkAddress{}
+
+	leaseFile := shared.VarPath("networks", network, "dnsmasq.leases")
+	if !shared.PathExists(leaseFile) {
+		return addresses, nil
+	}
+
+	dbInfo, err := networkLoadByName(s, network)
+	if err != nil {
+		return nil, err
+	}
+
+	content, err := ioutil.ReadFile(leaseFile)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, lease := range strings.Split(string(content), "\n") {
+		fields := strings.Fields(lease)
+		if len(fields) < 5 {
+			continue
+		}
+
+		// Parse the MAC
+		mac := networkGetMacSlice(fields[1])
+		macStr := strings.Join(mac, ":")
+
+		if len(macStr) < 17 && fields[4] != "" {
```

Comment:
@stgraber what does this bit do? Is this extracting the MAC from the DHCPv6 DUID? If so it might be worth making this into a generic parsing function for a single row of the leases file as it could be used in the bridged nic device's IP filtering functionality to discover existing allocations. I wasn't aware that the DHCPv6 lease contained the MAC so it will be great if it does.

Reply:
I think that's what it does, yeah, we've had that logic for a while in the lease parsing code I copy/pasted, I remember struggling to get a MAC out of the IPv6 lease and having to read through the dnsmasq code to figure out where it may end up.

Reply:
@stgraber interesting, I tested it this morning with DHCPv6 and it didn't pick it up. Perhaps some DHCPv6 clients generate DUIDs containing the MAC, but apparently not the one in Ubuntu 18.04.